src/tr2/game/inventory_ring/control.c

```diff
@@ -0,0 +1,1062 @@
+        char string[64];
+        sprintf(string, fmt, qty);
+        Overlay_MakeAmmoString(string);
+        g_Inv_ItemText[1] = Text_Create(64, -56, string);
+        Text_AlignBottom(g_Inv_ItemText[1], true);
+        Text_CentreH(g_Inv_ItemText[1], true);
+    }
+}
+
+static void M_ShowAmmoQuantity(const char *const fmt, const int32_t qty)
+{
+    if (!g_SaveGame.bonus_flag) {
+        M_ShowItemQuantity(fmt, qty);
+    }
+}
+
+static void M_Construct(const INVENTORY_MODE mode)
+{
+    if (mode == INV_TITLE_MODE) {
+        g_Inv_OptionObjectsCount = TITLE_RING_OBJECTS;
+        if (g_GameFlow.gym_enabled) {
+            g_Inv_OptionObjectsCount++;
+        }
+        m_VersionText = Text_Create(-20, -18, g_TR2XVersion);
+        Text_AlignRight(m_VersionText, 1);
+        Text_AlignBottom(m_VersionText, 1);
+        Text_SetScale(
+            m_VersionText, TEXT_BASE_SCALE * 0.5, TEXT_BASE_SCALE * 0.5);
+    } else {
+        g_Inv_OptionObjectsCount = OPTION_RING_OBJECTS;
+        Text_Remove(m_VersionText);
+        m_VersionText = NULL;
+    }
+
+    for (int32_t i = 0; i < 8; i++) {
+        g_Inv_ExtraData[i] = 0;
+    }
+
+    for (int32_t i = 0; i < g_Inv_MainObjectsCount; i++) {
+        INV_ITEM *const inv_item = g_Inv_MainList[i];
+        inv_item->meshes_drawn = inv_item->meshes_sel;
+        inv_item->current_frame = 0;
+        inv_item->goal_frame = 0;
+        inv_item->anim_count = 0;
+        inv_item->x_rot_pt = 0;
+        inv_item->x_rot = 0;
+        inv_item->y_rot = 0;
+        inv_item->y_trans = 0;
+        inv_item->z_trans = 0;
+        if (inv_item->object_id == O_PASSPORT_OPTION) {
+            inv_item->object_id = O_PASSPORT_CLOSED;
+        }
+    }
+
+    for (int32_t i = 0; i < g_Inv_OptionObjectsCount; i++) {
+        INV_ITEM *const inv_item = g_Inv_OptionList[i];
+        inv_item->current_frame = 0;
+        inv_item->goal_frame = 0;
+        inv_item->anim_count = 0;
+        inv_item->x_rot_pt = 0;
+        inv_item->x_rot = 0;
+        inv_item->y_rot = 0;
+        inv_item->y_trans = 0;
+        inv_item->z_trans = 0;
+        if (inv_item->object_id == O_PASSPORT_OPTION) {
+            inv_item->object_id = O_PASSPORT_CLOSED;
+        }
+    }
+
+    g_Inv_MainCurrent = 0;
+
+    if (g_GymInvOpenEnabled && mode == INV_TITLE_MODE
+        && !g_GameFlow.load_save_disabled && g_GameFlow.gym_enabled) {
+        g_Inv_OptionCurrent = 3; // TODO: don't hardcode me
+        g_SoundOptionLine = 0;
+    } else {
+        g_Inv_OptionCurrent = 0;
+        g_GymInvOpenEnabled = 0;
+        g_SoundOptionLine = 0;
+    }
+}
+
+static void M_End(INV_RING *const ring)
+{
+    M_RemoveAllText();
+    if (ring->list != NULL) {
+        INV_ITEM *const inv_item = ring->list[ring->current_object];
+        if (inv_item != NULL) {
+            Option_Shutdown(inv_item);
+        }
+    }
+    Memory_Free(ring);
+    Output_UnloadBackground();
+    Music_Stop();
```

Comment:
Changing this to the following resolves the music issue, not sure if you want to tackle it on develop or just fix it here as it's unreleased.
```
if (ring->mode == INV_TITLE_MODE) {
    Music_Stop();
}
```

Reply:
Thanks,I've addressed it here.